### ISSUE TITLE
How can i rewrite images in this code

### ISSUE THREAD

**Issue by Love4Taylor** [pagespeed/ngx_pagespeed#1281]
pagespeed EnableFilters rewrite_images
pagespeed EnableFilters convert_jpeg_to_webp
pagespeed EnableFilters convert_to_webp_lossless

`<script>gallery_set=[{type:"image",image:"https://www.tokototech.com/wp-content/uploads/2016/09/54604325_p0.jpg",thmb:"https://www.tokototech.com/wp-content/uploads/2016/09/54604325_p0-130x130.jpg",alt:"",title:"",description:"",titleColor:"#000000",descriptionColor:"#000000"},{type:"image",image:"https://www.tokototech.com/wp-content/uploads/2016/09/57090239_p0.jpg",thmb:"https://www.tokototech.com/wp-content/uploads/2016/09/57090239_p0-130x130.jpg",alt:"",title:"",description:"",titleColor:"#000000",descriptionColor:"#000000"},{type:"image",image:"https://www.tokototech.com/wp-content/uploads/2016/09/afafasfsa.jpg",thmb:"https://www.tokototech.com/wp-content/uploads/2016/09/afafasfsa-130x130.jpg",alt:"",title:"",description:"",titleColor:"#000000",descriptionColor:"#000000"},{type:"image",image:"https://www.tokototech.com/wp-content/uploads/2016/09/65-1604061H020.jpg",thmb:"https://www.tokototech.com/wp-content/uploads/2016/09/65-1604061H020-130x130.jpg",alt:"",title:"",description:"",titleColor:"#000000",descriptionColor:"#000000"},]
jQuery(document).ready(function(){jQuery('html').addClass('hasPag');jQuery('.custom_bg').remove();jQuery('body').fs_gallery({fx:'fade',fit:'no_fit',slide_time:3300,autoplay:1,show_controls:1,slides:gallery_set});jQuery('.fs_share').click(function(){jQuery('.fs_fadder').removeClass('hided');jQuery('.fs_sharing_wrapper').removeClass('hided');jQuery('.fs_share_close').removeClass('hided');});jQuery('.fs_share_close').click(function(){jQuery('.fs_fadder').addClass('hided');jQuery('.fs_sharing_wrapper').addClass('hided');jQuery('.fs_share_close').addClass('hided');});jQuery('.fs_fadder').click(function(){jQuery('.fs_fadder').addClass('hided');jQuery('.fs_sharing_wrapper').addClass('hided');jQuery('.fs_share_close').addClass('hided');});jQuery('.fs_controls').addClass('up_me');jQuery('.fs_title_wrapper ').addClass('up_me');jQuery('.close_controls').click(function(){if(jQuery(this).hasClass('open_controls')){jQuery('.fs_controls').removeClass('hide_me');jQuery('.fs_title_wrapper ').removeClass('hide_me');jQuery('.fs_thmb_viewport').removeClass('hide_me');jQuery('header.main_header').removeClass('hide_me');jQuery(this).removeClass('open_controls');}else{jQuery('header.main_header').addClass('hide_me');jQuery('.fs_controls').addClass('hide_me');jQuery('.fs_title_wrapper ').addClass('hide_me');jQuery('.fs_thmb_viewport').addClass('hide_me');jQuery(this).addClass('open_controls');}});});</script>`

**Comment by huibaolin** [comment]
Your images are loaded by JS. To optimize them, you can use the "OptimizeForBandwidth" mode [1]. To make the optimized cache-able by your downstream CDN, you can use the "AllowVaryOn" option [2]. Please see the following documents for details.

[1] https://developers.google.com/speed/pagespeed/module/optimize-for-bandwidth
[2] https://developers.google.com/speed/pagespeed/module/reference-image-optimize#AllowVaryOn

**Comment by oschaaf** [comment]
Currently nxg_pagespeed is not able to parse javascript, so it is not able to rewrite the urls in your script block.  It is possible to optimize these images in-place though, see:
https://developers.google.com/speed/pagespeed/module/system#ipro

Looking at a sample image, it looks like that currently is not working because (I think) the CDN is caching the unoptimized version:
https://www.tokototech.com/wp-content/uploads/2016/09/65-1604061H020.jpg 

```
cache-control:public, max-age=14400
cf-cache-status:HIT
cf-ray:2e88f9526e112c24-AMS
content-encoding:gzip
content-security-policy:block-all-mixed-content
content-type:image/jpeg
date:Mon, 26 Sep 2016 19:03:05 GMT
etag:"eb8641701118d21:0"
expires:Mon, 26 Sep 2016 23:03:05 GMT
last-modified:Mon, 26 Sep 2016 16:17:12 GMT
server:cloudflare-nginx
status:200
vary:Accept-Encoding
x-content-type-options:nosniff
x-frame-options:SAMEORIGIN
x-xss-protection:1; mode=block
```

Busting the cache by modifying the querystring, I end up with:
https://www.tokototech.com/wp-content/uploads/2016/09/65-1604061H020.jpg?a

```
accept-ranges:bytes
cache-control:public, max-age=14400
cf-cache-status:MISS
cf-ray:2e88f9196e262c24-AMS
content-encoding:gzip
content-security-policy:block-all-mixed-content
content-type:image/jpeg
date:Mon, 26 Sep 2016 19:02:57 GMT
etag:W/"PSA-RwUBeBft-a"
expires:Mon, 26 Sep 2016 23:02:57 GMT
server:cloudflare-nginx
status:200
vary:Accept-Encoding
x-content-type-options:nosniff
x-frame-options:SAMEORIGIN
x-xss-protection:1; mode=block
```

The unoptimized version is 665KB, the optimized version is 407KB.  
The in-place optimized versions can be recognized because they have `etag:W/"PSA-..."`.  
Note that you can't see the sizes in the headers because of the chunked transfer-encoding that is applied. 

In short, these things would be worth looking in to:
- Purging the CDN cache, to refresh the unoptimized resources with in-place optimized  versions. 
- The chunked transfer-encoding might be caused by the gzip compression that is applied. I think it would be wise not to gzip the image content-types, because that is probably not helping at all.
